server/chat-plugins/repeats.ts

```diff
@@ -0,0 +1,166 @@
+/**
+ * Chat plugin for repeating messages in chat
+ * Based on bot functionality from Kid A and Expecto Botronum
+ * @author Annika
+ */
+
+export interface RepeatedPhrase {
+	phrase: string;
+	/** interval in milliseconds */
+	interval: number;
+}
+
+/** roomid:phrase:timeout map */
+export const repeats = new Map<RoomID, Map<string, NodeJS.Timeout | null>>();
+
+export function hasRepeat(room: BasicRoom, phrase: string) {
+	return !!repeats.get(room.roomid)?.get(phrase);
+}
+
+export function addRepeat(room: BasicRoom, repeat: RepeatedPhrase) {
+	if (!room.settings.repeats) room.settings.repeats = [];
+	room.settings.repeats.push(repeat);
+	room.saveSettings();
+	runRepeat(room, repeat);
+}
+
+export function removeRepeat(room: BasicRoom, phrase: string) {
+	if (!room.settings.repeats) return;
+	room.settings.repeats = room.settings.repeats.filter(repeat => repeat.phrase !== phrase);
+	room.saveSettings();
+
+	if (!repeats.has(room.roomid)) return;
+	const oldInterval = repeats.get(room.roomid)?.get(phrase);
+	if (oldInterval) clearInterval(oldInterval);
+	repeats.get(room.roomid)?.set(phrase, null);
+}
+
+function runRepeat(room: BasicRoom, repeat: RepeatedPhrase) {
+	if (!repeats.has(room.roomid)) {
+		repeats.set(room.roomid, new Map<string, NodeJS.Timeout | null>());
```

Comment:
```suggestion
		repeats.set(room.roomid, new Map());
```

In theory the type can be inferred, right?